In [1]:
import pandas as pd
import numpy as np

In [11]:
df = pd.read_csv('wb_extract.csv')

Overall Accuracy

In [12]:
print(df['GPT-OG-Correct'].value_counts()/len(df))
print(df['Gemini-OG-Correct'].value_counts()/len(df))
print(df['LLAMA-OG-Correct'].value_counts()/len(df))

True     0.838384
False    0.161616
Name: GPT-OG-Correct, dtype: float64
True     0.883838
False    0.116162
Name: Gemini-OG-Correct, dtype: float64
True     0.835227
False    0.164773
Name: LLAMA-OG-Correct, dtype: float64


In [13]:
print(df['GPT-New-Correct'].value_counts()/len(df))
print(df['Gemini-New-Correct'].value_counts()/len(df))
print(df['LLAMA-New-Correct'].value_counts()/len(df))

True     0.886364
False    0.113636
Name: GPT-New-Correct, dtype: float64
True     0.945707
False    0.054293
Name: Gemini-New-Correct, dtype: float64
True     0.901515
False    0.098485
Name: LLAMA-New-Correct, dtype: float64


Accuracy by stereotype


In [19]:
print(df[df['Stereotype'] == 'pro']['GPT-OG-Correct'].value_counts()/len(df[df['Stereotype'] == 'pro']))
print(df[df['Stereotype'] == 'pro']['Gemini-OG-Correct'].value_counts()/len(df[df['Stereotype'] == 'pro']))
print(df[df['Stereotype'] == 'pro']['LLAMA-OG-Correct'].value_counts()/len(df[df['Stereotype'] == 'pro']))

True     0.931818
False    0.068182
Name: GPT-OG-Correct, dtype: float64
True     0.954545
False    0.045455
Name: Gemini-OG-Correct, dtype: float64
True     0.954545
False    0.045455
Name: LLAMA-OG-Correct, dtype: float64


In [21]:
print(df[df['Stereotype'] == 'anti']['GPT-OG-Correct'].value_counts()/len(df[df['Stereotype'] == 'anti']))
print(df[df['Stereotype'] == 'anti']['Gemini-OG-Correct'].value_counts()/len(df[df['Stereotype'] == 'anti']))
print(df[df['Stereotype'] == 'anti']['LLAMA-OG-Correct'].value_counts()/len(df[df['Stereotype'] == 'anti']))

True     0.744949
False    0.255051
Name: GPT-OG-Correct, dtype: float64
True     0.813131
False    0.186869
Name: Gemini-OG-Correct, dtype: float64
True     0.715909
False    0.284091
Name: LLAMA-OG-Correct, dtype: float64


In [22]:
print(df[df['Stereotype'] == 'pro']['GPT-New-Correct'].value_counts()/len(df[df['Stereotype'] == 'pro']))
print(df[df['Stereotype'] == 'pro']['Gemini-New-Correct'].value_counts()/len(df[df['Stereotype'] == 'pro']))
print(df[df['Stereotype'] == 'pro']['LLAMA-New-Correct'].value_counts()/len(df[df['Stereotype'] == 'pro']))

True     0.941919
False    0.058081
Name: GPT-New-Correct, dtype: float64
True     0.993687
False    0.006313
Name: Gemini-New-Correct, dtype: float64
True     0.97601
False    0.02399
Name: LLAMA-New-Correct, dtype: float64


In [23]:
print(df[df['Stereotype'] == 'anti']['GPT-New-Correct'].value_counts()/len(df[df['Stereotype'] == 'anti']))
print(df[df['Stereotype'] == 'anti']['Gemini-New-Correct'].value_counts()/len(df[df['Stereotype'] == 'anti']))
print(df[df['Stereotype'] == 'anti']['LLAMA-New-Correct'].value_counts()/len(df[df['Stereotype'] == 'anti']))

True     0.830808
False    0.169192
Name: GPT-New-Correct, dtype: float64
True     0.897727
False    0.102273
Name: Gemini-New-Correct, dtype: float64
True     0.82702
False    0.17298
Name: LLAMA-New-Correct, dtype: float64


Accuracy by sex


In [24]:
df['Pronoun']

0       him
1       her
2       him
3       her
4       him
       ... 
1579    him
1580    her
1581    him
1582    her
1583    him
Name: Pronoun, Length: 1584, dtype: object

In [26]:
def sex_from_pronoun(pronoun):
    if pronoun.lower() in ['him','his','he']:
        return 'Male'
    elif pronoun.lower() in ['she','her','hers']:
        return 'Female'

In [27]:
df['Sex'] = df['Pronoun'].apply(lambda x: sex_from_pronoun(x))

In [32]:
print(df[df['Sex'] == 'Male']['GPT-OG-Correct'].value_counts()/len(df[df['Sex'] == 'Male']))
print(df[df['Sex'] == 'Male']['Gemini-OG-Correct'].value_counts()/len(df[df['Sex'] == 'Male']))
print(df[df['Sex'] == 'Male']['LLAMA-OG-Correct'].value_counts()/len(df[df['Sex'] == 'Male']))

True     0.845088
False    0.154912
Name: GPT-OG-Correct, dtype: float64
True     0.904282
False    0.095718
Name: Gemini-OG-Correct, dtype: float64
True     0.870277
False    0.129723
Name: LLAMA-OG-Correct, dtype: float64


In [29]:
print(df[df['Sex'] == 'Female']['GPT-OG-Correct'].value_counts()/len(df[df['Sex'] == 'Female']))
print(df[df['Sex'] == 'Female']['Gemini-OG-Correct'].value_counts()/len(df[df['Sex'] == 'Female']))
print(df[df['Sex'] == 'Female']['LLAMA-OG-Correct'].value_counts()/len(df[df['Sex'] == 'Female']))

True     0.831646
False    0.168354
Name: GPT-OG-Correct, dtype: float64
True     0.863291
False    0.136709
Name: Gemini-OG-Correct, dtype: float64
True     0.8
False    0.2
Name: LLAMA-OG-Correct, dtype: float64


In [30]:
print(df[df['Sex'] == 'Male']['GPT-New-Correct'].value_counts()/len(df[df['Sex'] == 'Male']))
print(df[df['Sex'] == 'Male']['Gemini-New-Correct'].value_counts()/len(df[df['Sex'] == 'Male']))
print(df[df['Sex'] == 'Male']['LLAMA-New-Correct'].value_counts()/len(df[df['Sex'] == 'Male']))

True     0.900504
False    0.099496
Name: GPT-New-Correct, dtype: float64
True     0.963476
False    0.036524
Name: Gemini-New-Correct, dtype: float64
True     0.929471
False    0.070529
Name: LLAMA-New-Correct, dtype: float64


In [31]:
print(df[df['Sex'] == 'Female']['GPT-New-Correct'].value_counts()/len(df[df['Sex'] == 'Female']))
print(df[df['Sex'] == 'Female']['Gemini-New-Correct'].value_counts()/len(df[df['Sex'] == 'Female']))
print(df[df['Sex'] == 'Female']['LLAMA-New-Correct'].value_counts()/len(df[df['Sex'] == 'Female']))

True     0.872152
False    0.127848
Name: GPT-New-Correct, dtype: float64
True     0.927848
False    0.072152
Name: Gemini-New-Correct, dtype: float64
True     0.873418
False    0.126582
Name: LLAMA-New-Correct, dtype: float64


Table creation

True     0.872152
False    0.127848
Name: GPT-New-Correct, dtype: float64
True     0.927848
False    0.072152
Name: Gemini-New-Correct, dtype: float64
True     0.873418
False    0.126582
Name: LLAMA-New-Correct, dtype: float64


In [38]:
OverallOG = {'Type': 'Overall Original Accuracy','LLAMA':0.835227,'GPT':0.838384,'Gemini':0.883838}
OverallNew = {'Type': 'Overall New Accuracy','LLAMA':0.901515,'GPT':0.886364,'Gemini':0.945707}

ProSterOG = {'Type': 'Pro-Stereotype Original Accuracy','LLAMA':0.901515,'GPT':0.931818,'Gemini':0.954545}
AntiSterOG = {'Type': 'Anti-Stereotype Original Accuracy','LLAMA':0.715909,'GPT':0.744949,'Gemini':0.813131}
ProSterNew = {'Type': 'Pro-Stereotype New Accuracy','LLAMA':0.97601,'GPT':0.941919,'Gemini':0.993687}
AntiSterNew = {'Type': 'Anti-Stereotype New Accuracy','LLAMA':0.82702,'GPT':0.830808,'Gemini':0.897727}


MaleOG = {'Type': 'Male Original Accuracy','LLAMA':0.870277,'GPT':0.845088,'Gemini':0.904282}
FemaleOG= {'Type': 'Female Original Accuracy','LLAMA':0.8,'GPT':0.831646,'Gemini':0.863291}
MaleNew= {'Type': 'Male New Accuracy','LLAMA':0.929471,'GPT':0.900504,'Gemini':0.963476}
FemaleNew= {'Type': 'Female New Accuracy','LLAMA':0.873418,'GPT':0.872152,'Gemini':0.927848}

table = pd.DataFrame([OverallOG,OverallNew,ProSterOG,AntiSterOG,ProSterNew,AntiSterNew,MaleOG,FemaleOG,MaleNew,FemaleNew])

table.set_index('Type', inplace=True)

In [39]:
table

LLAMA       GPT    Gemini
Type                                                           
Overall Original Accuracy          0.835227  0.838384  0.883838
Overall New Accuracy               0.901515  0.886364  0.945707
Pro-Stereotype Original Accuracy   0.901515  0.931818  0.954545
Anti-Stereotype Original Accuracy  0.715909  0.744949  0.813131
Pro-Stereotype New Accuracy        0.976010  0.941919  0.993687
Anti-Stereotype New Accuracy       0.827020  0.830808  0.897727
Male Original Accuracy             0.870277  0.845088  0.904282
Female Original Accuracy           0.800000  0.831646  0.863291
Male New Accuracy                  0.929471  0.900504  0.963476
Female New Accuracy                0.873418  0.872152  0.927848

In [40]:
table.to_csv('accuracy_table.csv')